media/js/base/mozilla-traffic-cop.js

```diff
@@ -146,7 +146,21 @@ Mozilla.TrafficCop.prototype.generateRedirectUrl = function(url, querystring) {
     // conjure a random number between 1 and 100 (inclusive)
     var rando = Math.floor(Math.random() * 100) + 1;
 
-    url = url || window.location;
+    // make sure to disregard hash in original URL
```

Comment:
While I understand the premise that a hash might be unwanted in some test, I do wonder if we should err on the side of "if it's not broken, don't fix it"? 

For example, if a user is sent a URL like this: https://www.mozilla.org/en-US/firefox/channel/desktop/#beta, it should probably stick around, even if they go into some sort of experiment. If we have a case where it causes issues then maybe we could reconsider? I can't honestly decide what's best, it's a bit hard to say. But I'm happy to go with your judgement 😸